--- gcc-4.0.4/gcc/config/mips/linux.h-orig	2008-03-23 09:24:20.000000000 -0500
+++ gcc-4.0.4/gcc/config/mips/linux.h	2008-03-23 09:24:45.000000000 -0500
@@ -115,7 +115,7 @@
     %{!ibcs: \
       %{!static: \
         %{rdynamic:-export-dynamic} \
-        %{!dynamic-linker:-dynamic-linker /lib/ld.so.1}} \
+        %{!dynamic-linker:-dynamic-linker /lib/ld-uClibc.so.0}} \
         %{static:-static}}}"
 
 #undef SUBTARGET_ASM_SPEC
